app/src/main/java/org/thoughtcrime/securesms/notifications/DefaultMessageNotifier.java

```diff
@@ -505,24 +511,40 @@ private NotificationState constructNotificationState(@NonNull  Context context,
           continue;
         }
       }
+
+      // If this is a message request from an unknown user..
       if (messageRequest) {
         body = SpanUtil.italic(context.getString(R.string.message_requests_notification));
+
+      // If we received some manner of notification but Session is locked..
       } else if (KeyCachingService.isLocked(context)) {
         body = SpanUtil.italic(context.getString(R.string.MessageNotifier_locked_message));
+
+      // ----- All further cases assume we know the contact and that Session isn't locked -----
+
+      // If this is a notification about a multimedia message from a contact we know about..
       } else if (record.isMms() && !((MmsMessageRecord) record).getSharedContacts().isEmpty()) {
         Contact contact = ((MmsMessageRecord) record).getSharedContacts().get(0);
         body = ContactUtil.getStringSummary(context, contact);
+
+      // If this is a notification about a multimedia message which contains no text but DOES contain a slide deck with at least one slide..
       } else if (record.isMms() && TextUtils.isEmpty(body) && !((MmsMessageRecord) record).getSlideDeck().getSlides().isEmpty()) {
         slideDeck = ((MediaMmsMessageRecord)record).getSlideDeck();
         body = SpanUtil.italic(slideDeck.getBody());
+
+      // If this is a notification about a multimedia message, but it's NOT (perhaps because it's a `MediaMmsMessageRecord` that for some
+      // reason returns false), AND it contains a slide deck with at least one slide..
       } else if (record.isMms() && !record.isMmsNotification() && !((MmsMessageRecord) record).getSlideDeck().getSlides().isEmpty()) {
         slideDeck = ((MediaMmsMessageRecord)record).getSlideDeck();
```

Comment:
```suggestion
      } else if (record.isMms() && !record.isMmsNotification() && !((MmsMessageRecord) record).getSlideDeck().getSlides().isEmpty()) {
        // This is a notification about a multimedia message, but it's NOT (perhaps because it's a `MediaMmsMessageRecord` that for some
        // reason returns false), AND it contains a slide deck with at least one slide..
        slideDeck = ((MediaMmsMessageRecord)record).getSlideDeck();
```

I would say put these comments inside the block, rather than at the end of an unrelated block...

but... this particular comment doesn't add any information that isn't plainly obvious from the if statement itself.

Reply:
There's arguments both ways. From my perspective the point of the comment is to lower cognitive load when reading through the code to figure out what's going on - and as a comment it of course never makes its way into the compiled code, it's just for fallible humans like me. I feel that the reading flow works better to have the comment before the logic.